File: scala-pro/src/main/scala/systems/vostok/algo/search/SequentialSearch.scala
package systems.vostok.algo.search

import org.slf4j.LoggerFactory
import systems.vostok.generator.SequenceGenerator
import systems.vostok.performance.TimeChecker

import util.control.Breaks._

object SequentialSearch extends App {
  val logger = LoggerFactory.getLogger(classOf[TimeChecker])

  val WORST_CASE_NUMBER = 1000000000 // 1_000_000_000
  val sequence = SequenceGenerator.generateSimpleSequence(WORST_CASE_NUMBER)

  val timeChecker = new TimeChecker()
  timeChecker.printCurrent()

  breakable {
    for(value <- sequence) {
      if(value == WORST_CASE_NUMBER) {
        timeChecker.printCurrent()
        logger.info("Find value: {}", value)
        break
      }
    }
  }
}

File: groovy-pro/src/main/java/systems/vostok/dump/search/SequentialSearch.java
package systems.vostok.dump.search;

class SequentialSearch {
    public static void main(String[] args) {
        int[] array = {1,2,6,8,7,9,15,61,71,84};

        System.out.println(search(15, array));
        System.out.println(search(10, array));
    }

    static int search(int value, int[] array) {
        for(int arrayValue: array) {
            if(arrayValue == value) {
                return arrayValue;
            }
        }
        return -1;

    }
}
